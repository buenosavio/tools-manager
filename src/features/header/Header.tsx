import {
  ButtonAdd,
  HeaderMain,
  InputSearch,
  TitlePage,
  SubTitlePage,
  SpanSearchTags,
  SearchSection
} from './Header.styles'
import ModalNewCard from '../form/ModalNewCard'
import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { toolActions } from '../../store/tools/duck'

const Header = () => {
  const dispatch = useDispatch()
  const [modalOpen, setModalOpen] = useState(false)

  return (
    <HeaderMain>
      <TitlePage>VUTTR</TitlePage>
      <SubTitlePage>Very Useful Tools to Remember</SubTitlePage>
      <SearchSection>
        <div>
          <InputSearch
            type="text"
            name="search"
            placeholder="Search..."
            onChange={e => dispatch(toolActions.search(e.target.value))}
          />
          <input
            type="checkbox"
            onClick={() => dispatch(toolActions.toggleIsOnlyTag())}
          />
          <SpanSearchTags>search in tags only</SpanSearchTags>
        </div>
        <ButtonAdd onClick={() => setModalOpen(true)}>Add</ButtonAdd>
        {modalOpen && <ModalNewCard setOpenModal={setModalOpen} />}
      </SearchSection>
    </HeaderMain>
  )
}

export default Header
