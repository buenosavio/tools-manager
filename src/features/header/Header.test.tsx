import Header from './Header'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, renderHook, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

describe('Header component should be renderer ok', () => {
  const initialState = { tools: [], isOnlyTag: true, loading: true, search: '' }
  const mockStore = configureStore()
  let store = mockStore(initialState)

  test('renders correctly by default UI', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Title should be rendered on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const titlePage = screen.getByText('VUTTR')
    expect(titlePage).toBeInTheDocument()
  })

  test('Sub-Title should be rendered on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const subTitlePage = screen.getByText('Very Useful Tools to Remember')
    expect(subTitlePage).toBeInTheDocument()
  })

  test('Button-Add should be rendered on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const buttonAdd = screen.getByRole('button', { name: /Add/i })
    expect(buttonAdd).toBeInTheDocument()
  })

  test('Button-Add should be enabled on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const buttonAdd = screen.getByRole('button', { name: /Add/i })
    expect(buttonAdd).toBeEnabled()
  })

  test('Search input should be rendered and enable on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const searchInput = screen.getAllByPlaceholderText('Search...')
    expect(searchInput).toBeTruthy()
  })

  test('Toggle should be enable on screen', () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    const tagCheckbox = screen.getByRole('checkbox')
    expect(tagCheckbox).toBeEnabled()
  })

  test('Modal should be boolean', () => {
    const fn = jest.fn()
    let tree = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    // eslint-disable-next-line testing-library/await-async-query
    const button = tree.root.findByType('button')
    button.props.onClick()

    expect(fn.mock.calls.length).toBe(0)
  })
})
