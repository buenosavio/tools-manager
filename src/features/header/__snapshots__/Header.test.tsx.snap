// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be renderer ok renders correctly by default UI 1`] = `
<header
  className="sc-bczRLJ cdzGpq"
>
  <h1
    className="sc-gsnTZi iHDvAZ"
  >
    VUTTR
  </h1>
  <h3
    className="sc-dkzDqf ICVwW"
  >
    Very Useful Tools to Remember
  </h3>
  <div
    className="sc-hKMtZM idGchu"
  >
    <div>
      <input
        className="sc-eCYdqJ gSofyM"
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <input
        onClick={[Function]}
        type="checkbox"
      />
      <span
        className="sc-jSMfEi icURsM"
      >
        search in tags only
      </span>
    </div>
    <button
      className="sc-gKXOVf cCHAos"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</header>
`;
